src/test/directives/live_test.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * @license
+ * Copyright (c) 2019 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+/// <reference path="../../../node_modules/@types/mocha/index.d.ts" />
+/// <reference path="../../../node_modules/@types/chai/index.d.ts" />
+
+import {live} from '../../directives/live.js';
+import {render} from '../../lib/render.js';
+import {html} from '../../lit-html.js';
+
+const assert = chai.assert;
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+class LiveTester extends HTMLElement {
+  _x?: string;
+  _setCount = 0;
+
+  get x(): string|undefined {
+    return this._x;
+  }
+
+  set x(v: string|undefined) {
+    this._x = v;
+    this._setCount++;
+  }
+}
+customElements.define('live-tester', LiveTester);
+
+suite('live', () => {
+  let container: HTMLDivElement;
+
+  setup(() => {
+    container = document.createElement('div');
+  });
+
+  suite('properties', () => {
+    test('updates an externally set property', () => {
```

Comment:
Might be nice to validate what happens without `live`, at the least to aid readability.

Reply:
done